articles/quickstart/native/ionic-react/01-login.md

```diff
@@ -0,0 +1,262 @@
+
+:::note
+Local Storage should be considered **transient** in a Capacitor app, as the operating system may recover disk space from local storage if it is running low.
+
+[some blurb here about the SPA SDK extensible cache, when it's ready]
+:::
+
+:::panel Checkpoint
+Now that you have configured `Auth0Provider`, run your application to verify that the SDK is initializing correctly, and your application is not throwing any errors related to Auth0.
+:::
+
+## Add Login to Your Application
+
+In a Capacitor application, the [Capacitor's Browser plugin](https://capacitorjs.com/docs/apis/browser) should be used to perform a redirect to the Auth0 [Universal Login Page](https://auth0.com/universal-login). Use the `buildAuthorizeUrl` function to get the URL to redirect the user.
+
+:::note
+If you have used `auth0-react` before, you might be familiar with the `loginWithRedirect` function that redirects your SPA to the Auth0 Universal Login Page so that your users can authenticate, before returning to your app.
+
+This is done internally by setting `window.location.href` to the correct URL, but that isn't desireable for a Capacitor application, as it would use the default browser application on the user's device, rather than the system browser component appropriate for the platform. This means the user would leave your application and could make for a detrimental user experience.
+:::
+
+Add a new file `LoginButton.tsx` with the following code:
+
+```js
+import { useAuth0 } from "@auth0/auth0-react";
+import { Browser } from "@capacitor/browser";
+import { IonButton } from "@ionic/react";
+
+const LoginButton: React.FC = () => {
+  const { buildAuthorizeUrl } = useAuth0();
+
+  const login = async () => {
+    // Ask auth0-react to build the login URL
+    const url = await buildAuthorizeUrl();
+
+    // Redirect using Capacitor's Browser plugin
+    await Browser.open({ url });
+  };
+
+  return <IonButton onClick={login}>Log in</IonButton>;
+};
+
+export default LoginButton;
+```
+
+<%= include('../../_includes/_auth0-react-classes-info.md') %>
+
+### Handling the callback
+
+Once a user has logged in using the Universal Login Page, they will be redirected back to your app using a URL with a custom URL scheme. The `appUrlOpen` event must be handled within your app, where `handleRedirectCallback` can be called to initialize the authentication state within the SDK.
+
+Add a handler to your `App` component that looks like the following:
+
+```js
+const { handleRedirectCallback } = useAuth0();
+
+useEffect(() => {
+  // Handle the 'appUrlOpen' event and call `handleRedirectCallback`
+  CapApp.addListener("appUrlOpen", async ({ url }) => {
+      if (
+        url.includes("state") &&
+        (url.includes("code") || url.includes("error"))
+      ) {
+        await handleRedirectCallback(url);
+      }
+      // Ignored on Android
+      await Browser.close();
+  });
+}, [handleRedirectCallback]);
+```
```

Comment:
I'm not too sure where this is supposed to go, is it supposed to go into the App component?  Is `CapApp` the component through import or is it declared in the same file?

Reply:
Good shout, can definitely do with some more info here 👍🏻

Reply:
I've added the import and tried to make the sample a bit more helpful when it comes to positioning. The sample was already headered with "Add a handler to your App component that looks like the following:" and I've reworded it slightly.